contrib/pkg/adm/managedns/enable.go

```diff
@@ -165,18 +168,39 @@ func (o *Options) Run(dynClient client.Client, args []string) error {
 
 	// Adding manageDNS to HiveConfig triggers a new rollout of the hiveadmission pods.
 	// To know when it's safe to proceed, we will list all hiveadmission pods and ensure
-	// they all have the expected env var set.
+	// they all reference the new configmap.
+
+	err = waitForHiveConfigToBeProcessed(dynClient)
+	if err != nil {
+		log.WithError(err).Fatal("gave up waiting for HiveConfig to be processed")
+	}
+
+	log.Debug("getting new configmap with updated dns data")
+	dnsConfigMapName, err := getUpdatedManagedDNSConfig(dynClient)
+	if err != nil {
+		log.WithError(err).Fatal("failed to get new configmap name")
+	}
+
+	if err := waitForHiveAdmissionPods(dynClient, dnsConfigMapName); err != nil {
```

Comment:
Do we need to do this advanced wait any more? Can we simply wait for the hiveadmission deployment to roll out?